dwave/system/testing.py

```diff
@@ -31,30 +31,101 @@
 import concurrent.futures
 import numpy as np
 
-class MockDWaveSampler(dwave.system.samplers.DWaveSampler):
-    """Mock sampler modeled after DWaveSampler that can be used for tests.
+def _dnx_graph(topology_type, shape, nodelist, edgelist):
+    """Converts DWaveSampler's structure to a Chimera, Pegasus or Zephyr NetworkX graph.
+       
+       This code mirrors the to_networkx_graph method in dwave.system.samplers.DWaveSampler.
```

Comment:
Not sure why would you want to copy the code from `DWaveSampler.to_networkx_graph()` here. Why not just copy the function object from the `DWaveSampler` class? (see code change suggestion below)

If you want to keep topology shape checks, perhaps move them upstream to `DWaveSampler.to_networkx_graph`?

Reply:
to_networkx_graph is sort of a misnomer. What we really want is to_dwave_networkx_graph(). In the end, I thought it was simpler just to extract the code as single public function, since it has uses independent of any particular sampler. However, the code should really be located in the dwave_networkx repository in my opinion.